onboarding/main.go

```diff
@@ -122,3 +163,23 @@ func convertRsaPublicKeyAsPemStr(publicKey *rsa.PublicKey) (string, error) {
 
 	return string(publicKeyPem), nil
 }
+
+func getStorageClusterMetadata(ctx context.Context, operatorNamespace string, clientset *kubernetes.Clientset) (metav1.PartialObjectMetadata, error) {
+	var storageClusterMetadataList metav1.PartialObjectMetadataList
+	storageClusterGVKPath := fmt.Sprintf("/apis/%s/%s/namespaces/%s/%s", "ocs.openshift.io", "v1", operatorNamespace, "storageclusters")
+	instance, err := clientset.RESTClient().Get().AbsPath(storageClusterGVKPath).DoRaw(ctx)
+	if err != nil {
+		klog.Errorf("failed to get storage cluster metadata: %v", err.Error())
+		return metav1.PartialObjectMetadata{}, fmt.Errorf("failed to get storage cluster metadata: %v", err)
```

Comment:
I did not notice it in prev review but you are returning a copy of the metadata. 
There is no reason to do that, hold the metadata as a pointer and in all of the error cases return nil. 

```suggestion
		return nil, fmt.Errorf("failed to get storage cluster metadata: %v", err)
```